src/postings/recorder.rs

```diff
@@ -113,12 +117,24 @@ impl Recorder for NothingRecorder {
         buffer_lender: &mut BufferLender,
         serializer: &mut FieldSerializer<'_>,
         heap: &MemoryArena,
+        docid_map: Option<&DocidMapping>,
     ) {
         let buffer = buffer_lender.lend_u8();
         self.stack.read_to_end(heap, buffer);
-        // TODO avoid reading twice.
-        for doc in VInt32Reader::new(&buffer[..]) {
-            serializer.write_doc(doc as u32, 0u32, &[][..]);
+        //TODO avoid reading twice.
+        if let Some(docid_map) = docid_map {
+            let mut docids = VInt32Reader::new(&buffer[..])
```

Comment:
We should probably double check that this allocation has no impact...

Reply:
I can count the number of docids, which are recorded in order to correctly resize, but I think it would be best to have here also a datastructure which allows random access to retrieve the data in order without any additional allocation. This would mean to replace the ExpUnrolledList with something closer to BlockedBitpacker, but I'm not yet sure about the exact datastructure.

Reply:
Ah I forgot to append to this comment. Since recorder data is quite sparse in most cases (zipfs), memory overhead should not be large for the intermediate data to reorder for most cases, expect peaks for very common words like "and" etc, which could include all docids. ExpUnrolledList is a good choice for infrequent words, but for these very frequent words, the memory peak could be costly to reorder on some datasets. I didn't see any issues on the webserver data set, I'll test with the wiki dataset.

Reply:
Sorry I meant it as opposed to using the borrow buffer trick.

Reply:
done. It's not possible in the other cases, but as mentioned in the comments below it should have no noticeable impact on indexing speed.